### PR TITLE
Fix stake pool max num delegates

### LINKED ISSUE #3069: Add max delegate numbers per stake pool checking when creating stake pool for provider
The max delegate number per stake pool for providers should be limited by the smart contract's global config's `max_delegates`. The code shown in the link lakes the max_delegates checking.
https://github.com//0chain/0chain/blob/sprint-1.12/code/go/0chain.net/smartcontract/storagesc/blobber.go#L992

https://0chain.slack.com/archives/C03SYPVBZ33/p1704093526581099?thread_ts=1703966519.164179&cid=C03SYPVBZ33

### PR DESCRIPTION
**Pull request by Jayashsatolia403**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by peterlimg — APPROVED** [review]
LGTM

File: code/go/0chain.net/smartcontract/minersc/delegate_pools.go [hunk 1]
```diff
@@ -47,12 +48,74 @@ func (_ *MinerSmartContract) getStakePoolAdapter(pType spenum.Provider, provider
 		return mn, common.NewErrorf("get_stake_pool",
 			"unexpected DB error: %v", err)
 	}
+
+	if err := mn.save(balances); err != nil {
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/minersc/delegate_pools.go]: you can't save in get method

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/minersc/delegate_pools.go (hunk 1 above)
Comment (Jayashsatolia403) [reply code/go/0chain.net/smartcontract/minersc/delegate_pools.go]: Accidentally added it. Fixed.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  |   |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  |   |
| zwalletcli  |   |
| 0chain  | fix/max-delegates  |
| blobber  |   |
| authorizer  |   |
| 0box  |   |
| 0dns  |   |

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**